homeassistant/components/pjlink/media_player.py

```diff
@@ -85,50 +89,73 @@ def __init__(self, host, port, name, encoding, password):
         self._muted = False
         self._pwstate = MediaPlayerState.OFF
         self._current_source = None
-        with self.projector() as projector:
-            if not self._name:
-                self._name = projector.get_name()
-            inputs = projector.get_inputs()
-        self._source_name_mapping = {format_input_source(*x): x for x in inputs}
-        self._source_list = sorted(self._source_name_mapping.keys())
+        self._source_name_mapping = {}
+        self._source_list = []
+        self._available = self._initialize()
+
+    def _force_off(self):
+        self._pwstate = MediaPlayerState.OFF
+        self._muted = False
+        self._current_source = None
+
+    def _initialize(self):
+        try:
+            with self.projector() as projector:
+                if not self._name:
+                    self._name = projector.get_name()
+                inputs = projector.get_inputs()
+                self._source_name_mapping = {format_input_source(*x): x for x in inputs}
```

Comment:
Please only include the lines that can raise within the try... except block. We can move these lines below.

Reply:
I've moved these lines.